src/snowflake/snowpark/modin/pandas/general.py

```diff
@@ -394,29 +394,163 @@ def merge_ordered(
 
 
 @snowpark_pandas_telemetry_standalone_function_decorator
-@pandas_module_level_function_not_implemented()
 @_inherit_docstrings(pandas.merge_asof, apilink="pandas.merge_asof")
 def merge_asof(
     left,
     right,
-    on=None,
-    left_on=None,
-    right_on=None,
+    on: str | None = None,
+    left_on: str | None = None,
+    right_on: str | None = None,
     left_index: bool = False,
     right_index: bool = False,
-    by=None,
-    left_by=None,
-    right_by=None,
-    suffixes=("_x", "_y"),
-    tolerance=None,
+    by: str | list[str] | None = None,
+    left_by: str | None = None,
+    right_by: str | None = None,
+    suffixes: Suffixes = ("_x", "_y"),
+    tolerance: int | Timedelta | None = None,
     allow_exact_matches: bool = True,
     direction: str = "backward",
-) -> DataFrame:  # noqa: PR01, RT01, D200
+):  # noqa: PR01, RT01, D200
     """
     Perform a merge by key distance.
+
+    This is similar to a left-join except that we match on nearest key rather than equal keys.
+    Both DataFrames must be sorted by the key. For each row in the left DataFrame:
```

Comment:
the documentation should for pd modular should be provided in the same way as how other api is provided, right? otherwise, once @sfc-gh-joshi removes those file, the doc will be removed also. 
@sfc-gh-joshi can you help here to see how to configure the docstring for general.py?

Reply:
Good callout -- All the general `pd` APIs are done this way (e.g. `pivot`, `merge`), so I don't think this should block this PR but definitely something worth noting.

Reply:
Yeah. It seems we may not have such capability today yet. I am following up on this with @sfc-gh-joshi offline on this.